forest/cli/src/cli/send_cmd.rs

```diff
@@ -7,9 +7,76 @@ use forest_json::message::json::MessageJson;
 use forest_rpc_client::{mpool_push_message, wallet_default_address};
 use fvm_shared::{address::Address, econ::TokenAmount, message::Message, METHOD_SEND};
 use num::BigInt;
+use rust_decimal::prelude::*;
+use rust_decimal_macros::dec;
 
 use super::{handle_rpc_err, Config};
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct FILAmount {
+    pub value: TokenAmount,
+}
+
+impl FromStr for FILAmount {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let suffix_idx = s.rfind(char::is_numeric);
+        let (val, suffix) = match suffix_idx {
+            Some(idx) => s.split_at(idx + 1),
+            None => anyhow::bail!("failed to parse string: {}", s),
+        };
+
+        let suffix = suffix.trim().to_lowercase();
+        let multiplier = match suffix.strip_suffix("fil").map(str::trim).unwrap_or(&suffix) {
+            "atto" | "a" => dec!(1e0),
+            "femto" => {
+                dec!(1e3)
+            }
+            "pico" => {
+                dec!(1e6)
+            }
+            "nano" => {
+                dec!(1e9)
+            }
+            "micro" => {
+                dec!(1e12)
+            }
+            "milli" => {
+                dec!(1e15)
+            }
+            "" => {
+                dec!(1e18)
+            }
+            _ => {
+                anyhow::bail!("unrecognized suffix: {}", suffix);
+            }
+        };
+
+        // string length check to match Lotus logic
+        if val.chars().count() > 50 {
+            anyhow::bail!("string length too large: {}", val.chars().count());
+        }
```

Comment:
This check could be moved up in the code. There's no need to parse the suffix if we're just gonna bail due to the input length.